homeassistant/components/tautulli/config_flow.py

```diff
@@ -0,0 +1,179 @@
+"""Config flow for Tautulli."""
+from __future__ import annotations
+
+from typing import Any
+
+from pytautulli import PyTautulli, exceptions
+import voluptuous as vol
+
+from homeassistant.components.tautulli.coordinator import TautulliDataUpdateCoordinator
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_MONITORED_USERS, DATA_KEY_COORDINATOR, DEFAULT_NAME, DOMAIN
+
+
+class TautulliConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tautulli."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return TautulliOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            info, error = await self.validate_input(user_input)
+            if error is None:
+                await self.async_set_unique_id(info.pms_identifier)
+                self._abort_if_unique_id_configured(updates=user_input)
+                return self.async_create_entry(
+                    title=DEFAULT_NAME,
+                    data=user_input,
+                )
+            errors["base"] = error
+
+        user_input = user_input or {}
+        data_schema = {
+            vol.Required(CONF_API_KEY, default=user_input.get(CONF_API_KEY, "")): str,
+            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+            vol.Optional(CONF_PORT, default=user_input.get(CONF_PORT, "8181")): str,
+            vol.Optional(CONF_PATH, default=user_input.get(CONF_PATH, "")): str,
+            vol.Optional(CONF_SSL, default=user_input.get(CONF_SSL, False)): bool,
```

Comment:
This can be simplified by asking for the URL instead of all parts of the URL

Reply:
It seems that port is not automatically set to 8181 if we specify a url instead. This is less user friendly. The default port could just be listed in the description. Ideas?

Reply:
I could make the default url `http://[HOST]:8181`

Reply:
The port is a part od the URL, we can do all 5 fields but, 2 seemed more userfrendly?